libflux/flux-core/src/formatter/mod.rs

```diff
@@ -1027,12 +1203,23 @@ impl Formatter {
             if !src.is_empty() {
                 // Preserve the exact literal if we have it
                 self.write_string(src);
+                if src.contains("\n") {
+                    self.safe_to_reindent = false;
+                }
                 return;
             }
         }
         // Write out escaped string value
         self.write_rune('"');
         let escaped_string = self.escape_string(&n.value);
+
+        // if the string literal contains any newlines, then its not safe to `reindent` the string
+        // literal. Applying `reindent` to string literals with newlines can break them
+        // e.g. csv data
+        if escaped_string.contains("\n") {
+            self.unindent();
+            self.safe_to_reindent = false;
+        }
```

Comment:
This will prevent ever re-indenting a string literal that contains a newline, but it will also prevent reindention of other things that we _want_ to reident. Really, we want to be able to reindent anything except for a multiline string, but that is difficult to do after you've already formatted.

For example, I don't think this would work quite right:
```js
{
  b: 2,
  c: () => {
      // line 1
     a = 0
    return "a 
multi
line
string"    
    }
}
```
Property `c` is multi-line, the record literal would be multi-line, and we should reindent all the properties. But because `c` contains a string literal with a newline we won't, even though ideally we would want to indent the function body. Can you give this example a try and see what happens?

With that said, I think you've solved the most serious issue here, that the formatter wasn't idempotent. So I think it would be fine to include a test case like the one above, and file a bug to address it later (provided I'm correct in my suspicions). I think this case is difficult to solve given the current architecture of the formatter.

Reply:
Yes, you are right. In this case, the first closing brace is not appropriately indented. This is the output: 
```
 EXPECTED: 
 foo = {
    b: 2,
    c: () => {
        // line 1
        a = 0
        return "a
multi
line
string"
    },
} 
 OUTPUT: 
 foo = {
    b: 2,
    c: () => {
    // line 1
    a = 0

    return "a
multi
line
string"
},
} 
```

I do think that in this case it might be easier to isolate the issue of string literals versus other types into another smaller issue in order for that to be dealt with separately. I'll file that before I close this issue.